### ISSUE TITLE
Unable to get Request/Response

### ISSUE THREAD

**Issue by pankaj2joshi** [thepirat000/Audit.NET#97]
Hello,

I'm using audit.net library with .net core 2.0, this library don;t have support for saving/logging request body. Also as you know in .net core we can return IActionResult so in response it always store OkObjectResult instead of response I'm sending to client app. Please help.

**Comment by thepirat000** [comment]
oh you're right, the library should handle the OkResult, but I've tested with an OkObjectResult and didn't worked on net core 2.0. Will work on a fix for this.

**Comment by thepirat000** [comment]
This is fixed on version 11.0.2, please re-test and feel free to re-open if you're still having issues.

Regards

**Comment by thepirat000** [comment]
The problem with the request body is that, by design, it is treated as a forward-only stream that can be read only once.

So the audit filter will not able to read the request body without side-effects, unless another type of stream is used on the request body. A replacement of the default stream can be done via a global message handler (DelegatingHandler) but I think that's out of scope.

Any suggestions are very welcome!

**Comment by pankaj2joshi** [comment]
First of thank you for reply and fixing the issue. 

As per logging design complete request/response and other parameters are stored in file/database for each call, so in this case request part will be missing!  Manually adding "this.GetCurrentAuditScope()" in each request is little bit lengthy task, also we're unable to in which file or record response is saved, so we'll unable to link the request with response. 

Anyway we'll wait for updated build if we're able to store request/response in one go.

**Comment by thepirat000** [comment]
Ok, I finally implemented the `IncludeRequestBody` for AspNet Core on version 11.0.3. An important note was added to the README file and is copied also here:

When **IncludeResquestBody** is set to true and you are not using a `[FromBody]` parameter (i.e. reading the request body directly from the Request), make sure you enable rewind on the request body stream, otherwise the controller won't be able to read the request body since, by default, it's a forwand-only stream that can be read only once. You can enable rewind on your startup logic with the following code:
```c#
public void Configure(IApplicationBuilder app, IHostingEnvironment env)
{
    app.Use(async (context, next) => {  // <----
        context.Request.EnableRewind();
        await next();
    });
    
    app.UseMvc();
}
```

Hope it helps, please re-open if you are still having issues with this.

**Comment by pankaj2joshi** [comment]
Hello,

This is really amazing, I'll include latest code and test the entire code. Will get back to you if found any issue.

Your efforts are really appreciated, thanks a lot :)
